app/javascript/components/Header/index.js

```diff
@@ -1,40 +1,237 @@
-import React, { useContext } from 'react'
-
-import { Link } from 'react-router'
+import React, { useState, useContext } from 'react'
+import { Link, withRouter } from 'react-router'
 import * as R from 'ramda'
-import Popover from 'material-ui/Popover'
-import Menu from 'material-ui/Menu'
-import MenuItem from 'material-ui/MenuItem'
-import Divider from 'material-ui/Divider'
-import ArrowDropRight from 'material-ui/svg-icons/navigation-arrow-drop-right'
+import styled from 'styled-components'
 import { useMutation } from '@apollo/react-hooks'
+import { useTranslation } from 'react-i18next'
+
+import {
+  Dropdown,
+  Trigger,
+  Menu as GardenMenu,
+  Item,
+  Separator,
+  PreviousItem,
+  HeaderItem,
+  NextItem,
+} from '@zendeskgarden/react-dropdowns'
+import { Avatar } from '@zendeskgarden/react-avatars'
+import TranslationIcon from '@zendeskgarden/svg-icons/src/12/translation-exists-fill.svg'
+import { MD } from '@zendeskgarden/react-typography'
 
-import ContentIcon from 'material-ui/svg-icons/content/create'
 import UpdateUserOfficeMutation from '/mutations/UpdateUserOfficeMutation.gql'
 import { UserContext, FilterContext } from '/context'
-import { present } from '../../lib/utils'
-import s from './main.css'
 
-const styles = {
-  item: {
-    fontSize: 14,
-  },
+const NavigationText = styled.div`
+  color: #08c;
+`
+
+const UserDetails = styled.div`
+  margin-left: 10px;
+`
+
+const UserProfileContainer = styled.button`
+  display: flex;
+  border: none;
+  text-align: left;
+`
+
+const UserName = styled(MD)`
+  font-weight: bold;
+`
+
+const UserProfile = ({ currentUser, offices, handleOfficeSelect, location, router }) => {
```

Comment:
There is a lot going on in this file. It is actually housing 2 separate components. I would recommend moving the `UserProfile` component into it's own file for easy discovery and better separation of concerns.

In the `Header` component, have you considered using `{children}` instead of explicitly using `UserProfile`? 

The two components would be more composable after these suggestions imo.